PiT 23.1:  Unable to compile with latest prerelease `23.1.0.beta2`

### Issue
When compiling the demo with latest beta prerelease, it fails probably because tries to use `23.1.0`

```
[ERROR] Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ⇒ osgi.identity: (&(osgi.identity=com.vaadin.flow.data)(version>=6.0.0))
          ⇒ [com.vaadin.flow.data version=23.1.0]
              ⇒ osgi.wiring.package: (&(osgi.wiring.package=com.vaadin.flow.component)(version>=23.1.0)(version<=23.1.0))
                  ⇒ [com.vaadin.flow.server version=23.1.0]
                      ⇒ osgi.wiring.package: (&(osgi.wiring.package=com.vaadin.pro.licensechecker)(version>=1.4.0)(!(version>=2.0.0)))
    [org.eclipse.jetty.alpn.server version=9.4.35.v20201120]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.eclipse.jetty.alpn)(version>=1.1.3))
The following requirements are optional:
    [org.eclipse.jetty.websocket.server version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.registrar)
    [com.vaadin.flow.server version=23.1.0]
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.client)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.cpr)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.handler)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.cache)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.util)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.client.config)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.protocol)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.impl.client)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.client.methods)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.interceptor)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.client.protocol)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.auth)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.apache.http.impl.auth)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.atmosphere.client)(version>=2.7.0)(!(version>=3.0.0))(bundle-version>=2.7.3.slf4jvaadin4))
    [org.apache.felix.http.jetty version=4.1.4]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.service.log)(version>=1.3.0)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.service.metatype)(version>=1.1.0)(!(version>=2.0.0)))
      ⇒ osgi.serviceloader: (osgi.serviceloader=org.eclipse.jetty.io.ssl.ALPNProcessor$Server)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.service.cm)(version>=1.3.0)(!(version>=2.0.0)))
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.registrar)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.service.useradmin)(version>=1.1.0)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.service.event)(version>=1.2.0)(!(version>=2.0.0)))
    [org.eclipse.jetty.util version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.serviceloader: (osgi.serviceloader=org.eclipse.jetty.util.security.CredentialProvider)
    [com.helger.commons.ph-commons version=10.1.6]
      ⇒ osgi.serviceloader: (osgi.serviceloader=com.helger.commons.url.IURLProtocolRegistrarSPI)
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.junit.rules)
    [org.eclipse.jetty.http2.hpack version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.registrar)
    [org.eclipse.jetty.xml version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.serviceloader: (osgi.serviceloader=org.eclipse.jetty.xml.ConfigurationProcessorFactory)
    [com.vaadin.flow.data version=23.1.0]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.validation)(version>=2.0.1.Final))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.validation.metadata)(version>=2.0.1.Final))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.validation.constraints)(version>=2.0.1.Final))
    [ch.qos.logback.classic version=1.2.11]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.servlet.http)(version>=3.1.0)(!(version>=4.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.servlet)(version>=3.1.0)(!(version>=4.0.0)))
    [org.eclipse.jetty.server version=9.4.35.v20201120]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.eclipse.jetty.jmx)(version>=9.4.35)(!(version>=10.0.0)))
    [org.apache.commons.logging version=1.2.0]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.apache.log4j)(version>=1.2.15)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.servlet)(version>=2.1.0)(!(version>=3.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.apache.avalon.framework.logger)(version>=4.1.3)(version<=4.1.5))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.apache.log)(version>=1.0.1)(version<=1.0.1))
    [ch.qos.logback.core version=1.2.11]
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.codehaus.janino)
      ⇒ osgi.wiring.package: (osgi.wiring.package=javax.mail.internet)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.servlet.http)(version>=3.1.0)(!(version>=4.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.servlet)(version>=3.1.0)(!(version>=4.0.0)))
      ⇒ osgi.wiring.package: (osgi.wiring.package=org.codehaus.commons.compiler)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.fusesource.jansi)(version>=1.9.0)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (osgi.wiring.package=javax.mail)
    [org.eclipse.jetty.websocket.common version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.registrar)
    [org.eclipse.jetty.websocket.client version=9.4.35.v20201120]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.eclipse.jetty.xml)(version>=9.4.35)(!(version>=10.0.0)))
    [org.eclipse.jetty.security version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.serviceloader: (osgi.serviceloader=org.eclipse.jetty.security.Authenticator$Factory)
    [org.eclipse.jetty.client version=9.4.35.v20201120]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.eclipse.jetty.jmx)(version>=9.4.35)(!(version>=10.0.0)))
    [org.eclipse.jetty.websocket.api version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.serviceloader: (osgi.serviceloader=org.eclipse.jetty.websocket.api.extensions.Extension)
    [org.eclipse.jetty.servlet version=9.4.35.v20201120]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.eclipse.jetty.jmx)(version>=9.4.35)(!(version>=10.0.0)))
    [org.apache.log4j version=1.2.19]
      ⇒ osgi.wiring.package: (osgi.wiring.package=javax.jmdns)
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.mail)(version>=1.4.0)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=javax.mail.internet)(version>=1.4.0)(!(version>=2.0.0)))
      ⇒ osgi.wiring.package: (osgi.wiring.package=javax.jms)
    [org.eclipse.jetty.http version=9.4.35.v20201120]
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.processor)
      ⇒ osgi.extender: (osgi.extender=osgi.serviceloader.registrar)
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-export-maven-plugin:5.2.0:export (default) on project application: Unable to resolve <<INITIAL>>: missing requirement osgi.identity;filter:='(&(osgi.identity=com.vaadin.flow.data)(version>=6.0.0))' [caused by: Unable to resolve com.vaadin.flow.data version=23.1.0: missing requirement osgi.wiring.package;filter:='(&(osgi.wiring.package=com.vaadin.flow.component)(version>=23.1.0)(version<=23.1.0))';osgi.wiring.package='com.vaadin.flow.component' [caused by: Unable to resolve com.vaadin.flow.server version=23.1.0: missing requirement osgi.wiring.package;filter:='(&(osgi.wiring.package=com.vaadin.pro.licensechecker)(version>=1.4.0)(!(version>=2.0.0)))';osgi.wiring.package='com.vaadin.pro.licensechecker']] -> [Help 1]
```

### How to reproduce
- Clone the project  `https://github.com/vaadin/base-starter-flow-osgi.git`  (branch v23)
- Increase version number by running `mvn versions:set-property -Dproperty=vaadin.version -DnewVersion=23.1.0.beta2`
- Build by running `mvn clean install`
- See the error above

--- a/app/app.bndrun
+++ b/app/app.bndrun
@@ -10,5 +10,5 @@
 	bnd.identity;version='6.0';id='com.vaadin.flow.html.components',\
 	bnd.identity;version='6.0';id='com.vaadin.flow.push',\
 	bnd.identity;version='6.0';id='com.vaadin.flow.osgi',\
-	bnd.identity;version='1.1.1';id='com.vaadin.license-checker'
+	bnd.identity;version='1.4.1';id='com.vaadin.license-checker'
 

--- a/app/pom.xml
+++ b/app/pom.xml
@@ -16,7 +16,7 @@
     <properties>
         <bnd.version>5.2.0</bnd.version>
         <spifly.version>1.3.4</spifly.version>
-        <extra.bundles.dir>${project.build.directory}/extra-bundles</extra.bundles.dir>        
+        <extra.bundles.dir>${project.build.directory}/extra-bundles</extra.bundles.dir>
     </properties>
 
     <dependencies>
@@ -25,13 +25,13 @@
             <artifactId>project-base-osgi</artifactId>
             <version>${project.version}</version>
         </dependency>
-        
+
         <dependency>
             <groupId>com.vaadin</groupId>
             <artifactId>license-checker</artifactId>
-            <version>1.1.1</version>
+            <version>1.4.1</version>
         </dependency>
-        
+
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.framework</artifactId>
@@ -113,7 +113,7 @@
             <version>2.9.0</version>
         </dependency>
 
-        <!-- Those dependencies are need for OSGi container where this bundle 
+        <!-- Those dependencies are need for OSGi container where this bundle
             will be deployed. -->
 
         <dependency>
@@ -187,7 +187,7 @@
     </dependencies>
     <build>
         <plugins>
-            <!-- This dynamically calculates all the things we need to run 
+            <!-- This dynamically calculates all the things we need to run
                 our code. -->
             <plugin>
                 <groupId>biz.aQute.bnd</groupId>

